### PR TITLE
Disposals - Sort junctions can now hold a list of destinations they filter for

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR Do
Either through the var `sort_type_txt` for mappers, or using a destination tagger, sort junctions can be made to filter for multiple destinations. The tagger in particular can either remove an existing destination or add to the list for the targeted pipe.

Maps are touched up to convert the old `sortType` var into `sort_type_txt`.

## Why It's Good For The Game
This can vastly simplify disposal loops.

## Images of changes
https://user-images.githubusercontent.com/80771500/206768408-0cb18237-2610-40b9-8d35-3ab2a0a1e237.mp4

DisposalLists

## Testing
Tried disposals on mapload, added a junction that had multiple sort types, used the destination tagger to add or remove types from a junction.

Stuffed myself down a pipe to make sure I end up in the trash if no tags.

## Changelog
:cl:
add: Disposal sort junctions can filter for multiple destinations
add: Destination taggers can remove or add to the list of destinations a sort junction can filter for
/:cl:

### CODE REVIEW

File: code/modules/recycling/disposal.dm [hunk 1]
```diff
@@ -1052,15 +1054,34 @@
 
 	if(istype(I, /obj/item/destTagger))
 		var/obj/item/destTagger/O = I
+		var/tag = uppertext(GLOB.TAGGERLOCATIONS[O.currTag])
+		playsound(src.loc, 'sound/machines/twobeep.ogg', 100, 1)
+		if(O.currTag in sort_type)
+			sort_type -= O.currTag
+			to_chat(user, "<span class='notice'>Removed [tag] from filter.</span>")
+		else
+			sort_type += O.currTag
```
Comment (Contrabang) [inline code/modules/recycling/disposal.dm:1063]: hmm dumb question, but would this be better as `|=`?
```suggestion
			sort_type |= O.currTag
```

Comment (Vi3trice) [reply code/modules/recycling/disposal.dm:1063]: Works either way

**Comment by S34NW** [comment]
Looks good, just sort merge conflicts first

**Comment by S34NW** [comment]
``` _maps/map_files/Delta/delta.dmm:	name = "---Merge Conflict Marker---";
_maps/map_files/Delta/delta.dmm:	name = "---Merge Conflict Marker---";
_maps/map_files/Delta/delta.dmm:	name = "---Merge Conflict Marker---";
ERROR: Merge conflict markers detected in map, please resolve all merge failures!
_maps/map_files/Delta/delta.dmm:/obj/merge_conflict_marker{
_maps/map_files/Delta/delta.dmm:/obj/merge_conflict_marker{
_maps/map_files/Delta/delta.dmm:/obj/merge_conflict_marker{
ERROR: Merge conflict markers detected in map, please resolve all merge failures!```

File: code/modules/recycling/disposal.dm [hunk 2]
```diff
@@ -1054,15 +1056,34 @@
 
 	if(istype(I, /obj/item/destTagger))
 		var/obj/item/destTagger/O = I
+		var/tag = uppertext(GLOB.TAGGERLOCATIONS[O.currTag])
+		playsound(src.loc, 'sound/machines/twobeep.ogg', 100, 1)
```
Comment (Sirryan2002) [inline code/modules/recycling/disposal.dm:1060]:
```suggestion
		playsound(loc, 'sound/machines/twobeep.ogg', 100, 1)
```

**Review by Sirryan2002 — APPROVED** [review]
code looks good, merge conflicts taken care of

**Pull request merged**